splice_machine/src/main/java/com/splicemachine/derby/stream/function/NLJoinFunction.java

```diff
@@ -61,6 +68,8 @@
     protected boolean isOneRowInnerJoin;
     protected boolean hasMatch;
     protected List<Future<Pair<OperationContext, Iterator<ExecRow>>>> futures;
+    protected BlockingQueue<OperationContext> activeContexts;
```

Comment:
My understanding of activeContexts is that it has a one-to-one mapping to the list of futures. However, I only see that activeContexts get populated in loadBatch() but not in hasNext(), also I only see the context corresponding to the first left row removed from activeContexts in loadBatch(), should we remove the remaining ones as the rows get consumed in hasNext()?

Reply:
I think I overcomplicated the activeContexts a bit, I modified it to a list with all cloned contexts and closing all of them at the end.